File: Data Structures & Algorithms/two-integer-sum/submission-0.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> hm = new HashMap<>();

        for(int i = 0; i < nums.length; i++)    {
            int num = nums[i];
            int compl = target - num;
            if(hm.containsKey(compl))   {
                int[] sol = new int[2];
                int complIdx = hm.get(compl);
                sol[0] = complIdx;
                sol[1] = i;
                return sol;
            } else  {
                hm.put(num, i);
            }
        }

        return null;
    }
}

File: Data Structures & Algorithms/maximum-subarray/submission-0.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        maxSum = nums[0]
        currSum = 0

        for num in nums:
            currSum = max(currSum, 0) + num
            maxSum = max(currSum, maxSum)
        
        return maxSum